libmate-desktop/mate-colorsel.c

```diff
@@ -1022,13 +1025,15 @@ palette_paint (GtkWidget    *drawing_area,
 {
   gint focus_width;
   GtkAllocation allocation;
+  GdkRGBA bg;
 
   if (gtk_widget_get_window (drawing_area) == NULL)
     return;
 
   gtk_widget_get_allocation (drawing_area, &allocation);
 
-  gdk_cairo_set_source_color (cr, &gtk_widget_get_style (drawing_area)->bg[GTK_STATE_NORMAL]);
+  gtk_style_context_get_color (gtk_widget_get_style_context(drawing_area),GTK_STATE_FLAG_NORMAL, &bg);
```

Comment:
you're querying the foreground color, whereas the previous code used the background.

Reply:
What about making things simpler like this?
```diff
diff --git a/libmate-desktop/mate-colorsel.c b/libmate-desktop/mate-colorsel.c
index 148e429..7213c91 100644
--- a/libmate-desktop/mate-colorsel.c
+++ b/libmate-desktop/mate-colorsel.c
@@ -1025,15 +1025,15 @@ palette_paint (GtkWidget    *drawing_area,
 {
   gint focus_width;
   GtkAllocation allocation;
-  GdkRGBA bg;
+  gdouble color[4];
 
   if (gtk_widget_get_window (drawing_area) == NULL)
     return;
 
   gtk_widget_get_allocation (drawing_area, &allocation);
 
-  gtk_style_context_get_color (gtk_widget_get_style_context(drawing_area),GTK_STATE_FLAG_NORMAL, &bg);
-  gdk_cairo_set_source_rgba (cr, &bg);
+  palette_get_color (drawing_area, color);
+  cairo_set_source_rgba (cr, color[0], color[1], color[2], color[3]);
   cairo_rectangle (cr, 0, 0, allocation.width, allocation.height);
   cairo_fill (cr);
 
@@ -1221,26 +1221,6 @@ palette_change_color (GtkWidget         *drawing_area,
   g_free (current_colors);
 }
 
-static void
-override_background_color (GtkWidget *widget,
-                           GdkRGBA   *rgba)
-{
-  gchar          *css;
-  GtkCssProvider *provider;
-
-  provider = gtk_css_provider_new ();
-
-  css = g_strdup_printf ("* { color: %s;}",
-                         gdk_rgba_to_string (rgba));
-  gtk_css_provider_load_from_data (provider, css, -1, NULL);
-  g_free (css);
-
-  gtk_style_context_add_provider (gtk_widget_get_style_context (widget),
-                                  GTK_STYLE_PROVIDER (provider),
-                                  GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
-  g_object_unref (provider);
-}
-
 /* Changes the view color */
 static void
 palette_set_color (GtkWidget         *drawing_area,
@@ -1249,14 +1229,6 @@ palette_set_color (GtkWidget         *drawing_area,
 {
   gpointer pointer;
   gdouble *new_color = g_new (double, 4);
-  GdkRGBA  box_color;
-
-  box_color.red = color[0];
-  box_color.green = color[1];
-  box_color.blue = color[2];
-  box_color.alpha = 1;
-
-  override_background_color (drawing_area, &box_color);
 
   pointer = g_object_get_data (G_OBJECT (drawing_area), "color_set");
   if (!pointer || GPOINTER_TO_INT (pointer) == 0)
```

Reply:
Thank you for simplifying the code